src/modules/account/account.service.ts

```diff
@@ -0,0 +1,159 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { CreateAccountDto } from './dtos/create-account.dto';
+import { generate } from 'generate-password';
+import { UserRoles } from '@prisma/client';
+import { PrismaService } from '../prisma/prisma.service';
+import { UpdateAccountDto } from './dtos/update-account.dto';
+import { ExceptionsHandler } from '@nestjs/core/exceptions/exceptions-handler';
+
+@Injectable()
+export class AccountService {
+  constructor(private prisma: PrismaService) {}
+
+  async createAccount(data: CreateAccountDto, userId: number) {
+    try {
+      if (!userId) return null;
+
+      if (Object.keys(data).length === 0) {
+        throw new BadRequestException('No data provided.');
+      }
+
+      const password = generate({
+        length: 15,
+        numbers: true,
+      });
+
+      const accountInformation = {
+        ...data,
+        password,
+      };
+
+      const account = await this.prisma.account.create({
+        data: accountInformation,
+      });
+
+      const role = await this.prisma.role.findFirst({
+        where: { roleName: UserRoles.ADMIN },
+      });
+
+      await this.prisma.membership.create({
+        data: {
+          userId,
+          accountId: account.id,
+          roleId: role.id,
+          isConfirmed: true,
+        },
+      });
+
+      return account;
+    } catch (error) {
+      return new ExceptionsHandler(error.response);
```

Comment:
@Akeron44 return new ExceptionHandler -> I would suggest a different name that would describe the error exactly.
And like discussed, the service could just throw the errors. Controller can decide what to do with them (e.g. return the correct http response payload and status based on the error type)